### PR TITLE
Fix to Werkzeug ProxyFix; expose ProxyFix configuration items

### LINKED ISSUE #8062: CUSTOM_SECURITY_MANAGER not working in 0.34.0rc1
App routing is broken in the latest release with respect to Custom Security Manager.  Please deprecate this release.

### Expected results

When attempting to access the root url for Superset, this should redirect to the login page if the user is not logged in.  This redirect is failing to work when `AUTH_TYPE = AUTH_OID`

### Actual results

The app routes to `/superset/welcome`

#### How to reproduce the bug

1. Use a custom security manager, akin to https://stackoverflow.com/questions/54010314/using-keycloakopenid-connect-with-apache-superset#54024394
2. Deploy
3. Navigate to bare url

### Environment

(please complete the following information):

- superset version: 0.34.0rc1
- python version: 3.6
- node.js version: 10.0
- npm version: `npm -v`

### Checklist

Make sure these boxes are checked before submitting your issue - thank you!

- [x] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### Additional context

This works perfectly fine on `0.33.0rc1`, so I have had to downgrade.  Because of this issue, where the routing is not correct, I believe this issue should be resolved and 0.34.0rc1 should be pulled.

### LINKED ISSUE #8110: Investigate broken Custom Security Manager
I have made sufficient comments, and instead of sweeping this under the rug, it should be looked at - CSM is broken because redirection does not work properly - it works in 0.33.0rc1, but not 0.34!

https://github.com/apache/incubator-superset/issues/8062

### PR DESCRIPTION
**Pull request by ericandrewmeadows**
### CATEGORY

Choose one

- [x] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [x] Documentation

### SUMMARY
Custom Security Manager appeared to be broken when behind an AWS ALB, but when I dug further, the Werkzeug version was bumped.  This caused an invalid configuration for the proxy in that the host was not passed on.  Because of this, along with the change of the library path, and the inclusion of the additional library bump in accordance with https://github.com/apache/airflow/pull/5571, I have validated that the regression that prevented CSM from working in v0.33.0rc1 -> v0.34.0rc [1 & 2] has been resolved.

### TEST PLAN
I deployed internally the fix, and validated that the port is not being returned in the header for the destination after hitting the root IP of the Superset instance.

### ADDITIONAL INFORMATION

- [x] Has associated issue:
  - Fixes #8062 
  - Fixes #8110 
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by dpgaspar** [comment]
`black` config.py

**Comment by villebro** [comment]
Thanks for putting in the work to fix this @ericandrewmeadows . Once this gets merged I think we can start picking cherries for a `0.34.1`.

**Comment by ericandrewmeadows** [comment]
@dpgaspar / @villebro - what's the process for merging this?

**Comment by villebro** [comment]
LGTM, but I don't feel comfortable merging before @dpgaspar approves this, as he is more knowledgeable on this topic than I am.

**Comment by ericandrewmeadows** [comment]
I figured - yeah - it may have just slipped under his radar.

**Comment by ericandrewmeadows** [comment]
@dpgaspar - can this be merged?

**Comment by mistercrunch** [comment]
Few notes:
* do we know how this maps to previous default settings from the previous implementation?
* can we add an entry in `UPDATING.md` to tell people "Changes around a new version of ProxyFix in Werkzeuf, if you use config `ENABLE_PROXY_FIX = True`, you'll want to review the newly introduced `PROXY_FIX_CONFIG` config key and make sure it's set as expected for your environment"

**Comment by ericandrewmeadows** [comment]
1) I first fixed my instance, then @dpgaspar pointed out the defaults from Airflow so I followed that convention.  For me, I use the following settings:
```
ENABLE_PROXY_FIX = True
PROXY_FIX_CONFIG = {
    "x_for": 1,
    "x_proto": 0,
    "x_host": 1,
    "x_port": 0,
    "x_prefix": 0,
}
```
2) I'll add that to `UPDATING.md`

File: Untitled.ipynb [hunk 1]
```diff
@@ -0,0 +1,66 @@
+{
```
Comment (mistercrunch) [inline Untitled.ipynb]: ????? I think this file was not intended to be added here

**Review by mistercrunch — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: Untitled.ipynb (hunk 1 above)
Comment (ericandrewmeadows) [reply Untitled.ipynb]: 100% was not intended to be in there.

**Review by ericandrewmeadows — COMMENTED** [review]
@mistercrunch - Should be good to be merged.

**Review by mistercrunch — APPROVED** [review]

**Pull request merged**